src/ansys/turbogrid/core/multi_blade_row/multi_blade_row.py

```diff
@@ -21,1083 +21,531 @@
-from multiprocessing import Manager, Pool, Process, cpu_count
 import os
-import threading
-import time
+from pathlib import Path, PurePath
+import queue
+import traceback
+from typing import Optional
 
-from ansys.turbogrid.api import pyturbogrid_core
-from ansys.turbogrid.api.CCL.ccl_object_db import CCLObjectDB
+from ansys.turbogrid.api.pyturbogrid_core import PyTurboGrid
 from fabric import Connection
-from jinja2 import Environment, FileSystemLoader
-
-from ansys.turbogrid.core.launcher.launcher import launch_turbogrid
-from ansys.turbogrid.core.mesh_statistics import mesh_statistics
-import ansys.turbogrid.core.ndf_parser.ndf_parser as ndfp
-
-
-class MultiBladeRow:
-    """
-    Facilitates working on a multi blade row turbomachinery case using PyTurboGrid instances in parallel.
-    """
-
-    _working_dir: str = None
-    _case_directory: str = None
-    _case_ndf: str = None
-    _ndf_file_full_path: str = None
-    _results_directory: str = None
-    _ndf_parser: ndfp.NDFParser = None
-
-    _blade_rows_to_mesh: dict = None
-    _multi_process_count: int = 0
-    _blade_row_gsfs: dict = None
-    _blade_boundary_layer_offsets: dict = None
-    _blade_row_spanwise_counts: dict = None
-    _custom_blade_settings: dict = None
-
-    #: Number of decimal places to be used for values in the meshing reports.
-    report_stats_decimal_places = 3
-
-    #: The unit to be used for the angles in the meshing reports.
-    #: Use "rad" for angles in radian and "deg" for angles in degrees.
-    report_stats_angle_unit = "deg"
-
-    #: List of the quality measures to be reported for each blade row.
-    #: Permitted entries are:
-    #: "Connectivity Number", "Edge Length Ratio", "Element Volume Ratio",
-    #: "Maximum Face Angle", "Minimum Face Angle", "Minimum Volume",
-    #: "Orthogonality Angle", "Skewness"
-    report_mesh_quality_measures = [
-        "Edge Length Ratio",
-        "Minimum Face Angle",
-        "Minimum Volume",
-        "Orthogonality Angle",
-    ]
-
-    #: Set True to write tginit first before processing blade rows in parallel by reading the tginit
-    #: instead of reading ndf in parallel and writing separate tginit for each blade row.
-    write_tginit_first = False
 
-    #: When working in Ansys Labs, the number of attempts to be made for file transfer from
-    #: container to the local working directory results folder.
-    max_file_transfer_attempts = 20
+from ansys.turbogrid.core.launcher.launcher import launch_turbogrid, launch_turbogrid_container
+from ansys.turbogrid.core.multi_blade_row.single_blade_row import single_blade_row
+import ansys.turbogrid.core.ndf_parser.ndf_parser as ndf_parser
 
-    #: An alias for the TurboGridLocationType in pyturbogrid_core.PyTurboGrid.
-    TurboGridLocationType = pyturbogrid_core.PyTurboGrid.TurboGridLocationType
 
-    #: When working in Ansys Labs, the name with full path of the file containing the container key.
-    tg_container_key_file = ""
+class Helpers:
```

Comment:
@Acohen-work Many of these helpers are not directly connected to multi_blade_row. They should be put into their own class/file for clarity and re-use.